### PR TITLE
fix: escape keywords in TypeName

### LINKED ISSUE #37: Inconsistent escaping of raw path strings
Consider the following scenario:

File: `test/package/v1/TestMessage.proto`
```proto
syntax = "proto3";
package test.package.v1;
import "test/abstract/v1/target.proto";
message TestMessage {
    test.abstract.v1.Target target = 1;
}
```

File: `test/abstract/v1/Target.proto`
```proto
syntax = "proto3";
package test.abstract.v1;
message Target {
    bool example = 1;
}
```

When `pbjson` generates the `Serialize`/`Deserialize` impls, it will report the path to refer to `Target` as `super::super::abstract::v1::Target`. However, as `abstract` is a reserved keyword, this will fail to compile. The proper way to refer to this path is `super::super::r#abstract::v1::Target`. The `prost` generation correctly specifies this path.

### PR DESCRIPTION
**Pull request by mbrobbel**
An attempt to fix #37.

### CODE REVIEW

**Comment by tustvold** [comment]
Thank you for this, could we possibly get an integration style test in pbjson-test? :smile:

File: pbjson-test/src/lib.rs [hunk 1]
```diff
@@ -36,6 +36,16 @@ pub mod test {
...
+    pub mod escape {
+        include!(concat!(
+            env!("OUT_DIR"),
+            "/test.r#abstract.r#type.escape.rs"
+        ));
+        include!(concat!(
+            env!("OUT_DIR"),
+            "/test.r#abstract.r#type.escape.serde.rs"
+        ));
+    }
```
Comment (tustvold) [inline pbjson-test/src/lib.rs:48]: Could we possibly get some tests of serializing to/from these types?

File: pbjson-build/src/descriptor.rs [hunk 2]
```diff
@@ -68,7 +70,7 @@ impl TypeName {
 
     pub fn to_snake_case(&self) -> String {
```
Comment (tustvold) [inline pbjson-build/src/descriptor.rs:71]: Perhaps we should rename this to `to_snake_case_ident` or something, to make clearer that it performs escaping

**Review by tustvold — COMMENTED** [review]
Looking really nice, thank you. Could you please sign the [CLA](https://www.influxdata.com/legal/cla/) and I can get this in :smile:

File: pbjson-test/src/lib.rs (hunk 1 above)
Comment (mbrobbel) [reply pbjson-test/src/lib.rs:48]: [c608e46](https://github.com/influxdata/pbjson/pull/67/commits/c608e46ffe4ea6057d2e5eea9fca479e52640293)

**Comment by mbrobbel** [comment]
CLA ✅

**Review by tustvold — APPROVED** [review]

**Pull request merged**